packages/nimble-components/src/utilities/models/tracker.ts

```diff
@@ -0,0 +1,75 @@
+type ObjectFromList<T extends readonly string[]> = {
+    [K in T extends readonly (infer U)[] ? U : never]: boolean;
+};
+
+/**
+ * Generic Tracker which sets or resets provided flags
+ */
+export class Tracker<TrackedItemsList extends readonly string[]> {
+    private trackedItems: ObjectFromList<TrackedItemsList>;
+
+    public constructor(trackedItemsList: TrackedItemsList) {
+        type TrackedItems = typeof this.trackedItems;
+        this.trackedItems = {} as TrackedItems;
+        this.trackedItems = trackedItemsList.reduce<TrackedItems>(
+            (r, key): TrackedItems => {
+                return {
+                    ...r,
+                    [key]: false
+                };
+            },
+            this.trackedItems
+        );
+    }
+
+    public getTrackedItems(): ObjectFromList<TrackedItemsList> {
+        return { ...this.trackedItems };
+    }
+
+    public trackedItemState(
+        key: keyof ObjectFromList<TrackedItemsList>
+    ): boolean {
+        return this.trackedItems[key];
+    }
```

Comment:
Reading how it looks in source, can we rename this to `isTracked`? I think it matches the pattern of all/any/none-Tracked better for checking the tracked item status.